spring-cloud-function-web/src/main/java/org/springframework/cloud/function/web/FunctionHandlerMapping.java

```diff
@@ -0,0 +1,145 @@
+		implements InitializingBean {
+
+	public static final String FUNCTION = FunctionHandlerMapping.class.getName()
+			+ ".function";
+	public static final String CONSUMER = FunctionHandlerMapping.class.getName()
+			+ ".consumer";
+	public static final String SUPPLIER = FunctionHandlerMapping.class.getName()
+			+ ".supplier";
+	public static final String ARGUMENT = FunctionHandlerMapping.class.getName()
+			+ ".argument";
+	private final FunctionCatalog functions;
+
+	private final FunctionController controller;
+	
+	@Value("${spring.cloud.function.web.path:}")
+	private String prefix = ""; 
+
+	@Autowired
+	public FunctionHandlerMapping(FunctionCatalog catalog) {
+		this.functions = catalog;
+		setOrder(super.getOrder() - 5);
+		this.controller = new FunctionController();
+	}
+
+	@Override
+	public void afterPropertiesSet() {
+		super.afterPropertiesSet();
+		detectHandlerMethods(controller);
+		while (prefix.endsWith("/")) {
+			prefix = prefix.substring(0, prefix.length()-1);
+		}
+	}
+
+	@Override
+	protected HandlerMethod getHandlerInternal(HttpServletRequest request)
+			throws Exception {
+		HandlerMethod handler = super.getHandlerInternal(request);
+		if (handler == null) {
+			return null;
+		}
+		String path = (String) request
+				.getAttribute(HandlerMapping.PATH_WITHIN_HANDLER_MAPPING_ATTRIBUTE);
+		if (path == null) {
+			return handler;
+		}
+		if (findFunctionForGet(request, path) != null) {
+			return handler;
+		}
+		if (findFunctionForPost(request, path) != null) {
+			return handler;
+		}
+		return null;
+	}
+
+	private Object findFunctionForPost(HttpServletRequest request, String path) {
+		if (!request.getMethod().equals("POST")) {
+			return null;
+		}
+		path = path.startsWith("/") ? path.substring(1) : path;
+		Function<Object, Object> function = functions.lookupFunction(path);
+		if (function != null) {
+			request.setAttribute(FUNCTION, function);
+			return function;
+		}
+		Consumer<Object> consumer = functions.lookupConsumer(path);
+		if (consumer != null) {
+			request.setAttribute(CONSUMER, consumer);
+			return consumer;
+		}
+		return null;
+	}
+
+	private Object findFunctionForGet(HttpServletRequest request, String path) {
+		if (!request.getMethod().equals("GET")) {
+			return null;
+		}
+		path = path.startsWith("/") ? path.substring(1) : path;
+		StringBuilder builder = new StringBuilder();
+		String name = path;
+		String value = null;
+		for (String element : path.split("/")) {
+			if (builder.length()>0) {
+				builder.append("/");
+			}
+			builder.append(element);
+			name = builder.toString();
+			value = path.length() > name.length() ? path.substring(name.length() + 1)
+					: null;
+			Function<Object, Object> function = functions.lookupFunction(name);
+			if (function != null) {
+				request.setAttribute(FUNCTION, function);
+				request.setAttribute(ARGUMENT, value);
+				return function;
+			}
+		}
+		Supplier<Object> supplier = functions.lookupSupplier(path);
```

Comment:
for `GET` requests, should we give `Supplier` lookup precedence over `Function`? I guess at the bean name level, uniqueness is enforced across the set of both